### PR TITLE
Experience-9022: Add GHA workflow to deploy to trial frontend on label add

### LINKED ISSUE #9022: Add GHA workflow to deploy to trial frontend on label add
## Problem statement

We now have a GitHub Action to deploy trial frontends, and we can tweak it so it's more seamless to use in our implementation/design process.

## What you need to know

- [trial_frontend workflow](https://github.com/CDCgov/prime-reportstream/blob/master/.github/workflows/release_trial_frontend.yml)

## Acceptance criteria

- [ ] We can append a label for `trialfrontend(01|02|03)` on pull requests and trigger a trial frontend deployment

### PR DESCRIPTION
**Pull request by stephenkao**
This changeset updates the "Release Trial Frontend" workflow to be evented by labels rather than force-pushes to the trial frontend branches.  This'll help speed up our review process, and it'll help us determine if a particular trial frontend environment is already occupied by someone else.

The steps to use this new workflow:
- Open up a pull request (doesn't matter if it's draft or ready for review)
- Apply one of the following labels: `trialfrontend01`, `trialfrontend02`, `trialfrontend03`
- Wait for the github-actions bot to comment on deploy status
- Check out the linked trial frontend for your changes

## Linked Issues
- Fixes #9022

### CODE REVIEW

File: .github/actions/build-vars/action.yml
Comment (stephenkao) [inline .github/actions/build-vars/action.yml:52]: @JosiahSiegel Since we're going to be triggering trial frontend deployments from different branches (rather than `trialfrontend*` branches), these sections aren't really needed anymore.  Let me know if this is an issue!

File: frontend-react/src/App.tsx
Comment (stephenkao) [inline frontend-react/src/App.tsx:30]: Will remove before merge, but I wanted to make sure this was actually working on the trialfrontend01 environment :)

**Comment by stephenkao** [comment]
Blech, looks like an issue:

Screenshot 2023-04-06 at 15 03 53

Will take a look after my meetings!

File: .github/workflows/release_trial_frontend.yml
Comment (snesm) [inline .github/workflows/release_trial_frontend.yml:49]: We use [SHA hashes](https://devopsjournal.io/blog/2021/12/11/GitHub-Actions-Maturity-Levels) for all public actions

`peter-evans/find-comment@034abe94d3191f9c89d870519735beae326f2bdb`

File: .github/workflows/release_trial_frontend.yml
Comment (snesm) [inline .github/workflows/release_trial_frontend.yml:57]: `peter-evans/create-or-update-comment@3383acd359705b10cb1eeef05c0e88c056ea4666`

File: .github/workflows/release_trial_frontend.yml
Comment (snesm) [inline .github/workflows/release_trial_frontend.yml:68]: `actions/checkout@8f4b7f84864484a7bf31766abe9204da3cbe65b3`

File: .github/workflows/release_trial_frontend.yml
Comment (snesm) [inline .github/workflows/release_trial_frontend.yml:92]: `actions/checkout@8f4b7f84864484a7bf31766abe9204da3cbe65b3`

File: .github/workflows/release_trial_frontend.yml
Comment (snesm) [inline .github/workflows/release_trial_frontend.yml:112]: `peter-evans/find-comment@034abe94d3191f9c89d870519735beae326f2bdb`

File: .github/workflows/release_trial_frontend.yml
Comment (snesm) [inline .github/workflows/release_trial_frontend.yml:120]: `peter-evans/create-or-update-comment@3383acd359705b10cb1eeef05c0e88c056ea4666`

**Review by snesm — CHANGES_REQUESTED** [review]

File: .github/workflows/release_trial_frontend.yml
Comment (stephenkao) [reply .github/workflows/release_trial_frontend.yml:49]: Got it -- thanks for the context.  Will update here and in the other pull request :)

Comment (stephenkao) [reply .github/workflows/release_trial_frontend.yml]: Updated!  Feel free to take another look whenever!

**Review by jpandersen87 — APPROVED** [review]

**Review by penny-lischer — APPROVED** [review]
LGTM

**Comment by etanb** [comment]
How does one re-run the deploy to the trial frontend? Remove and replace the tag?

Can we update documentation as a part of this PR?

**Comment by stephenkao** [comment]
> How does one re-run the deploy to the trial frontend? Remove and replace the tag?
> 
> Can we update documentation as a part of this PR?

Yep, just remove the label and re-add it.  I definitely want to update the documentation as part of this changeset, but I wanted to get this in front of everyone's eyes first so we're agreed on the approach beforehand.

**Comment by stephenkao** [comment]
@jeremy-page @JosiahSiegel Mind taking another look at this when you get a chance?  Happy to keep the discussion going about the new workflow, but what's here right now is the proposed process.

**Comment by stephenkao** [comment]
Update: This is ON HOLD as we're still discussing the different deployment methods.

**Comment by stephenkao** [comment]
Closing this since we're not going to be moving forward with it for now.

**Pull request closed without merging**
